src/TextInput.tsx

```diff
@@ -37,14 +37,17 @@ const TextInput = React.forwardRef<HTMLInputElement, TextInputInternalProps>(
     // this class is necessary to style FilterSearch, plz no touchy!
     const wrapperClasses = classnames(className, 'TextInput-wrapper')
 
+    // temporarily fixes an issue using icon and block in tandem
+    const customStyles = IconComponent && block && {...sxProp, ...{display: 'flex'}}
+
     return (
       <TextInputWrapper
         block={block}
         className={wrapperClasses}
         contrast={contrast}
         disabled={disabled}
         hasIcon={!!IconComponent}
-        sx={sxProp}
+        sx={customStyles || sxProp}
```

Comment:
nitpick for readability

```js
 sx={{
  display: IconComponent && block ? 'flex' : 'inline-block',
  ...sxProp
}}
```

Reply:
Thanks @siddharthkp! Yes, was thinking along these lines initially too. I found that approach however, would force `inline-block` as a default style, overriding the `TextArea` css class styling (making this component more opinionated than it needed to be). 

The approach I had taken here only applies the override styling if the right conditions are met, leaving the default styling intact and reducing chance of regressions in future. At least that was the thinking 😅 .

Reply:
Ah I didn't realise there is already a display up the chain. Good thinking!

I'd also try this approach to set a value only when there is a value:

```
display: IconComponent && block ? 'flex' : undefined
```

Reply:
👍 ... In the end, there was a much simpler solution 😅. Would you mind reviewing the changes again please @siddharthkp?